tools/cli/commands/create.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for implementing the `datalab create` command."""
+
+import argparse
+import subprocess
+import tempfile
+
+import connect
+
+
+description=(
+"""{0} {1} creates a new Datalab instances running in a Google
+Compute Engine VM.
+
+This command also creates the 'datalab-network' network if necessary.
+
+By default, the command creates a persistent connection to the newly
+created instance. You can disable that behavior by passing in the
+'--no-connect' flag."""
+)
+
+
+_DATALAB_NETWORK = 'datalab-network'
+_DATALAB_NETWORK_DESCRIPTION = 'Network for Google Cloud Datalab instances'
+
+_DATALAB_FIREWALL_RULE = 'datalab-network-allow-ssh'
+_DATALAB_FIREWALL_RULE_DESCRIPTION = 'Allow SSH access to Datalab instances'
+
+_DATALAB_DEFAULT_DISK_SIZE_GB = 200
+_DATALAB_DISK_DESCRIPTION = (
+    'Persistent disk for a Google Cloud Datalab instance')
+
+_DATALAB_STARTUP_SCRIPT = """#!/bin/bash
+
+PERSISTENT_DISK_DEV="/dev/disk/by-id/google-datalab-pd"
+MOUNT_DIR="/mnt/disks/datalab-pd"
+MOUNT_CMD="mount -o discard,defaults ${PERSISTENT_DISK_DEV} ${MOUNT_DIR}"
+
+format_disk() {
+  mkfs.ext4 -F -E lazy_itable_init=0,lazy_journal_init=0,discard ${PERSISTENT_DISK_DEV}
+  ${MOUNT_CMD}
+}
+
+mount_disk() {
+  mkdir -p "${MOUNT_DIR}"
+  ${MOUNT_CMD} || format_disk
+  mount -o discard,defaults /dev/disk/by-id/google-datalab-pd /mnt/disks/datalab-pd || format_disk
+  chmod a+w "${MOUNT_DIR}"
+}
+
+mount_disk
+"""
+
+_DATALAB_CONTAINER_SPEC = """
```

Comment:
Similar comment about making this an external file, but still packaged within the tool.